srcs/commands/Join.cpp

```diff
@@ -90,5 +91,6 @@ int cmdJoin(Message &msg, Client *Client, std::map<std::string, Channel*> &chann
 		return (joinExistingServerWithoutKey(channels, msg.params[0], Client));
 	if (channels.size() > 0 && msg.params.size() == 2)
 		return (joinExistingServerWithKey(channels, msg.params[0], msg.params[1], Client));
+	Client->setNewChannel(getChannelName(msg.params[0]));
```

Comment:
this would add channnel when there is error. this line should be in join existingServerWithoutKey and joinExistingServerWithKey functions :)

Reply:
Resolved!